### PR TITLE
resmgr download: separate name and url

### LINKED ISSUE #780: resmgr download: allow specifying name independent of URL
```
ocrd resmgr download -n ocrd-detectron2-segment ...
ocrd resmgr list-installed -e ocrd-detectron2-segment
- All_X152.yaml @ data (https://layoutlm.blob.core.windows.net/tablebank/model_zoo/detection/All_X152/All_X152.yaml)
  Found at data on 2022-01-19 12:44:17.192397
- model_final_01ca85.pkl @ data (???)
  Found at data on 2022-01-20 23:29:42.287291
- model_final.pth @ data (https://layoutlm.blob.core.windows.net/tablebank/model_zoo/detection/All_X152/model_final.pth)
  Found at data on 2022-01-19 13:13:18.793976
- DLA_mask_rcnn_R_101_FPN_3x.yaml @ data (https://github.com/hpanwar08/detectron2/raw/master/configs/DLA_mask_rcnn_R_101_FPN_3x.yaml)
  Found at data on 2022-01-20 21:23:24.611257
```

This is not helpful. I would like to be able to set a `name` on the CLI, just as I can in the central `resource_list.yml`.

### PR DESCRIPTION
**Pull request by bertsky**
Fixes #780

### CODE REVIEW

**Comment by bertsky** [comment]
We should add a test for the `-n URL TOOL NAME` use-case as well (both as an unregistered resource and as URL-override).

File: ocrd/ocrd/cli/ocrd_tool.py [hunk 1]
```diff
@@ -96,15 +96,23 @@ def ocrd_tool_tool_description(ctx):
 @ocrd_tool_tool.command('list-resources', help="List tool's file resources")
 @pass_ocrd_tool
 def ocrd_tool_tool_list_resources(ctx):
-    Processor(None, ocrd_tool=ctx.json['tools'][ctx.tool_name],
-              list_resources=True)
+    class BashProcessor(Processor):
+        @property
+        def moduledir(self):
+            return os.path.dirname(ctx.filename)
```
Comment (kba) [inline ocrd/ocrd/cli/ocrd_tool.py:102]: But will this work with the `SHAREDIR=...` convention we use for `make install` of bashlib processors? In that case, the executable might be installed to `$HOME/.local/bin`, but the resources to `$HOME/.local/share/<projectname>`?

File: ocrd/ocrd/processor/base.py [hunk 2]
```diff
@@ -213,19 +205,50 @@ def resolve_resource(self, val):
...
-        List all resources found in the filesystem
+        List all resources found in the filesystem and matching content-type by filename suffix
+        """
+        mimetypes = get_processor_resource_types(None, self.ocrd_tool)
+        for res in list_all_resources(self.ocrd_tool['executable'], moduled=self.moduledir):
+            res = Path(res)
+            if not '*/*' in mimetypes:
+                if res.is_dir() and not 'text/directory' in mimetypes:
+                    continue
+                # if we do not know all MIME types, then keep the file, otherwise require suffix match
```
Comment (kba) [inline ocrd/ocrd/processor/base.py:233]: :+1:

File: ocrd_utils/ocrd_utils/os.py [hunk 3]
```diff
@@ -98,33 +100,45 @@ def list_all_resources(executable):
...
 
 def get_processor_resource_types(executable, ocrd_tool=None):
     """
-    Determine whether a processor has resource parameters that represent
-    directories (``has_dirs``), files (``has_files``) or neither.
+    Determine what type of resource parameters a processor needs.
 
-    Returns a pair ``(has_dir, has_files)``
+    Return a list of MIME types (with the special value `*/*` to
+    designate that arbitrary files or directories are allowed).
```
Comment (kba) [inline ocrd_utils/ocrd_utils/os.py:127]: That is much better than `has_dirs, has_files`.

**Review by kba — APPROVED** [review]

File: ocrd/ocrd/cli/ocrd_tool.py (hunk 1 above)
Comment (bertsky) [reply ocrd/ocrd/cli/ocrd_tool.py:102]: Yes, it does work – as long as the resources are in the same directory as the ocrd-tool.json file (which is the case for all existing bashlib processors we have). It does not matter where the executable itself is, or which directory the resources go to (venv or XDG or whatever `make install` does).

File: ocrd_utils/ocrd_utils/os.py (hunk 3 above)
Comment (bertsky) [reply ocrd_utils/ocrd_utils/os.py:127]: It's been quite a chase to get here, but yes, I think this makes sense now.

**Comment by bertsky** [comment]
Note: this warrants a new minor release, because the `resmgr download` CLI has changed slightly, and so has `ocrd_utils.get_processor_resource_types`. 

Documentation will have to be updated in various places:
- changelog here
- https://github.com/OCR-D/spec/pull/190
- models.md in spec

**Pull request closed without merging**